codegen/src/api/mod.rs

```diff
@@ -233,7 +237,7 @@ impl RuntimeGenerator {
     /// unique path, so that types with matching paths don't end up overwriting each other
     /// in the codegen. We ignore any types with generics; Subxt actually endeavours to
     /// de-duplicate those into single types with a generic.
-    fn ensure_unique_type_paths(metadata: &mut Metadata) {
+    pub fn ensure_unique_type_paths(metadata: &mut Metadata) {
```

Comment:
Why is this pub? (It's called on `new` above, so does exposing it help with something?)

Reply:
Ah I see!

Perhaps we should expose this function on our `Metadata` then instead, so that optionally one can make this be the case there too. It's a bit of a weird one; I pondered just having it run on the `Metadata` automatically but wasn't entirely comfortable with altering these things by default.

So yeah, let's add it on `Metadata` and then just call it here as well as wherever you need it :)